gordo_components/model/anomaly/diff.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+
+import os
+import json
+
+import numpy as np
+import pandas as pd
+
+from typing import Optional, Union
+from datetime import datetime, timedelta
+
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import MinMaxScaler
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from gordo_components import serializer
+from gordo_components.model.base import GordoBase
+
+
+class DiffBasedAnomalyDetector(BaseEstimator, GordoBase):
+    def __init__(
+        self,
+        base_estimator: BaseEstimator = PCA(),
+        scaler: TransformerMixin = MinMaxScaler(),
+    ):
+        """
+        Classifier which wraps a ``base_estimator`` and provides a diff error
+        based approach to anomaly detection.
+
+        It trains a ``scaler`` to the target **after** training, purely for
+        error calculations. The underlying ``base_estimator`` is trained
+        with the original, unscaled, ``y``.
+
+        Parameters
+        ----------
+        base_estimator: sklearn.base.BaseEstimator
+            The model to which normal ``.fit``, ``.predict`` methods will be used.
+        scaler: sklearn.base.TransformerMixn
+            Defaults to ``sklearn.preprocessing.MinMaxScaler``
+            Used for transforming model output and the original ``y`` to calculate
+            the difference/error in model output vs expected.
+        """
+        self.base_estimator = base_estimator
+        self.scaler = scaler
+
+    def __getattr__(self, item):
+        """
+        Treat this model as transparent into base_estimator unless
+        referring to something owned by this object
+        """
+        if item in self.__dict__:
+            return getattr(self, item)
+        else:
+            return getattr(self.base_estimator, item)
+
+    def get_metadata(self):
+        if isinstance(self.base_estimator, GordoBase):
```

Comment:
Is it not better to just check if it has get_metadata?

Reply:
Guess it could go both ways.. checking `GordoBase` ensures we know that it has `get_metadata` as well was _what_ we expect that to return.  But I'm not strongly attached